geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java

```diff
@@ -310,12 +310,16 @@ public long getConnectionID() {
     return this.connectionID;
   }
 
-  protected HandShake getHandShake() {
-    return handShake;
+  protected byte[] encryptBytes(byte[] messageBytes) throws Exception {
+    return handshake.getEncryptor().encryptBytes(messageBytes);
```

Comment:
Is there any risk of handshake not being set when this (or decryptBytes) is called?